src/state/slices/opportunitiesSlice/selectors/stakingSelectors.ts

```diff
@@ -79,6 +82,29 @@ export const selectUserStakingOpportunitiesById = createSelector(
 export const selectStakingOpportunitiesById = (state: ReduxState) =>
   state.opportunities.staking.byId
 
+export const selectStakingOpportunityByFilter = createSelector(
```

Comment:
```suggestion
export const selectStakingOpportunityByFilter = createDeepEqualOutputSelector(
```

Objects not referentially stable